Source/HTTP/WinHttp/winhttp_http_task.cpp

```diff
@@ -256,37 +259,36 @@ void winhttp_http_task::callback_status_data_available(
     _In_ void* statusInfo)
 {
     // Status information contains pointer to DWORD containing number of bytes available.
-    DWORD numBytes = *(PDWORD)statusInfo;
+    DWORD newBytesAvailable = *(PDWORD)statusInfo;
 
-    HC_TRACE_INFORMATION(HTTPCLIENT, "HCHttpCallPerform [ID %llu] WINHTTP_CALLBACK_STATUS_DATA_AVAILABLE numBytes=%d", pRequestContext->m_call->id, numBytes);
+    HC_TRACE_INFORMATION(HTTPCLIENT, "HCHttpCallPerform [ID %llu] WINHTTP_CALLBACK_STATUS_DATA_AVAILABLE newBytesAvailable=%d", pRequestContext->m_call->id, newBytesAvailable);
 
-    if (numBytes > 0)
+    if (newBytesAvailable > 0)
     {
-        http_internal_vector<uint8_t> bodyData;
-        bodyData.resize(numBytes);
+        size_t oldSize = pRequestContext->m_responseBuffer.size();
+        size_t newSize = oldSize + newBytesAvailable;
+        pRequestContext->m_responseBuffer.resize(newSize);
 
         // Read in body all at once.
         if (!WinHttpReadData(
             hRequestHandle,
-            &bodyData[0],
-            numBytes,
+            &pRequestContext->m_responseBuffer[oldSize],
+            newBytesAvailable,
             nullptr))
         {
             HC_TRACE_ERROR(HTTPCLIENT, "HCHttpCallPerform [ID %llu] WinHttpReadData errorcode %d", pRequestContext->m_call->id, GetLastError());
         }
-
-        pRequestContext->m_responseBuffer.append(&bodyData[0], static_cast<ULONG>(bodyData.size()));
     }
     else
     {
         // No more data available, complete the request.
         if (pRequestContext->m_responseBuffer.size() > 0)
         {
-            auto const& responseString = pRequestContext->m_responseBuffer.as_string();
-            if (responseString.length() > 0)
-            {
-                HCHttpCallResponseSetResponseString(pRequestContext->m_call, responseString.c_str());
-            }
+            char c[1] = { 0 };
+            pRequestContext->m_responseBuffer.insert(pRequestContext->m_responseBuffer.end(), &c[0], &c[0] + 1);
```

Comment:
why not just pRequestContext->m_responseBuffer.push_back(0)

Reply:
will do